doc/design/libsvm_format_column.md

```diff
@@ -0,0 +1,134 @@
+
+### LibSVM Format Column
+
+A **LibSVM format column** is a DBMS table column, whose value of each row is in the same format as each line in a **LibSVM format file**.
+
+For example, in the following table, `c1` is a LibSVM format column, while `c2` is not a LibSVM format column.
+
+```
++-------------------- +------+
+| c1                  | c2   |
++---------------------+------+
+| 0 1:2.1 3:4.5 5:6.0 |  2.8 |
+| 1 7:-3.2 9:2.3      |  3.1 |
++---------------------+------+
+```
+
+## Background
+
+SQLFlow extends SQL syntax to support the end-to-end machine learning pipeline. However, due to the variety of data formats, data pre-processing is usually a tough task for users. SQLFlow should do some automatic data pre-processing to make the end-to-end machine learning pipeline easier.
+
+Currently, SQLFlow only supports loading the column value which is a number, a string, or a number list in CSV format. We should support loading data from the LibSVM format column in SQLFlow, because the LibSVM format column is a common way to store the sparse data in DBMS, and there have been [some issues](https://github.com/sql-machine-learning/sqlflow/issues/2323) that require SQLFlow should support loading data from the LibSVM format column.
+
+There are 2 ways to support the LibSVM format column in SQLFlow:
+
+- `TO RUN` statements. It can support any user-defined data pre-processing. We can release a docker image to convert the data from the LibSVM format column to a dense tensor, create columns for each element of the dense tensor, and write the data into the result table. However, if the data is highly sparse, it would create too many columns in the result table, and consume lots of time and memories to do the transformation.
+- `COLUMN` clauses. The `COLUMN` clauses in SQLFlow support some commonly used feature columns, like `NUMERIC`, `CATEGORY_ID`, `EMBEDDING`, etc. If we want to support loading the data from the LibSVM format column, where would be 2 optional methods:
+    - Add a new `COLUMN` clause, like `LIBSVM` or `NUMERIC_KV`. It introduces complexity and disobeys the Occam's Razor principle.
+    - Extend the `NUMERIC` column clause to support columns in different data formats. Currently, we have supported CSV format implicitly in `NUMERIC` column clause: in feature derivation stage, we would infer whether the column data is in CSV format (see [here](https://github.com/sql-machine-learning/sqlflow/blob/3b70a0599beef573cd99f15dd41cc0a194634b75/pkg/ir/derivation.go#L146)), and convert the values in CSV format into tensors in Python side (see [here](https://github.com/sql-machine-learning/sqlflow/blob/develop/python/sqlflow_submitter/db.py#L159)). We can do the same implicit feature derivation and conversion for the LibSVM format column.
+    
+In conclusion, we would choose to extend the `NUMERIC` column to support the LibSVM format column in this design.
```

Comment:
Use `SPARSE` should be better, since we already describe sparse inputs using `SPARSE` syntax in column clause. e.g. `COLUMN NUMERIC(SPARSE(col, [10000], "\s"), shape)`

Reply:
There may be some problems if we use `SPARSE` :

- Both `NUMERIC` and `SPARSE` have the same parameter `shape`. It may be a conflict.
- I have tested that TensorFlow  does not support `SparseTensor` as the feature input of `tf.feature_column.numeric_column`. `SparseTensor` would only be used in the `EMBEDDING`, `CATEGORY_ID` layer.

Maybe we should redesign whether `SPARSE` should be used together with the `NUMERIC` column.